lib/publiccloud/azure.pm

```diff
@@ -72,14 +73,24 @@ sub find_img {
     my $container = $self->container;
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
     my $key = $self->get_storage_account_keys($storage_account);
-    my $cmd_show = "az storage blob show --account-key $key -o json " .
-      "--container-name '$container' --account-name '$storage_account' --name '$name' " .
-      '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"';
-    record_info('BLOB INFO', script_output($cmd_show, proceed_on_failure => 1));
 
-    my $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
+    $json = script_output("az storage blob show --account-key $key -o json " .
+          "--container-name '$container' --account-name '$storage_account' --name '$name' " .
+          '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"',
+        proceed_on_failure => 1);
+    record_info('BLOB INFO', $json);
+    eval {
+        $md5 = decode_azure_json($json)->{md5};
+    };
+    record_info('BLOB NOT-FOUND', "Cannot find blob $name. Need to upload it.\n$@") if ($@);
```

Comment:
```
if ($@) {
record_info('BLOB NOT-FOUND', "Cannot find blob $name. Need to upload it.\n$@") 
}
elif (!defined($md5) || $md5 !~ /^[a-fA-F0-9]{32}$/) {
        record_info('INVALID', "The blob $name does not have valid md5 field.");
        return $image;
 }
```

Reply:
two reasons : 
1. IMHO `!$md5` will cause "Use of uninitialized variable " perl exceptions in code which are quite annoying and distinctive when you parsing test log with hope to find root cause of the problem 
2. you have 2 if's one for "$@" and another for "!@$" IMHO it is looks better in single if/else